app/2.1.x/go.md

```diff
@@ -4,44 +4,128 @@ title: Go language support
+Guide introduces Go plugins and provides instructions on how to write and use
+them.
 
-## Prerequisites
+### New Concepts
 
-To use Go plugins:
+#### Go Plugin Server
 
-* You need the `go-pluginserver` executable.  The Kong distribution packages
-install it in `/usr/local/bin/`.  If you want it anywhere else, set the
-`go_pluginserver_exe` variable in the Kong configuration file to the full path.
-* Set the `go_plugins_dir` variable in the Kong configuration file to a valid directory.
-The default `"off"` value disables Go support.
-* Add compiled Go plugins to the directory specified in the previous step.
-* Add the plugin settings the usual way, either via Admin API, database or declarative file.
-Refer to the the plugin via its filename (without the `.so` suffix).
+In order for Kong to run Go plugins, it employs a sidecar process: the
+[go-pluginserver][go-pluginserver]. The role of this process is to dynamically
+load a Go plugin and execute its code on demand, based on communication
+with Kong (performed through a Unix domain socket). Kong is responsible for
+managing the life cycle of the go-pluginserver, and, as you will see below, it
+only needs to know where to find the go-pluginserver's executable to do so.
 
-To write your own Go plugins:
+#### The go-pdk
 
-* A Kong development environment as usual.
-* A Go development environment.
-* A `go-pluginserver` executable built with the same compiler and libraries.  The most direct way is to get it from source with `go get github.com/Kong/go-pluginserver`
-* Copy the new executable to `/usr/local/bin`, with `cp $(go env GOPATH)/bin/go-pluginserver /usr/local/bin`. Alternatively, you can specify the `go_pluginserver_exe` property with the path to the `go-pluginserver` binary
-* The Go-PDK package: `import "github.com/Kong/go-pdk"`.
+The [go-pdk][go-pdk] allows your Go plugins to access functionality provided
+by the [Kong PDK][kong-pdk]. In order to use it, you need to import add
+`"github.com/Kong/go-pdk"` to the list of packages imported by your plugin.
+The `kong` parameter received on the event handler methods is the entry point
+for the Go PDK functions. Most of these functions work the same as the corresponding
+functions in the Lua PDK.
 
-## Development
+See the [go-pdk](https://godoc.org/github.com/Kong/go-pdk) godoc page for the
+reference documentation of the Go PDK
 
-To write a Kong plugin in Go, you need to:
+### Prerequisites
+
+In order to use Go plugins, you need the following:
+
+1. A Go compiler, which can be obtained from your Operating System's package
+manager, or directly at the [Go Downloads page](https://golang.org/dl/);
+
+2. The `go-pluginserver` executable. By default, Kong will expect it in
+`/usr/local/bin`; however, this location can be customized with the `go_pluginserver_exe`
+property in the Kong configuration file - or via environment variables, like
+any other Kong configuration property;
```

Comment:
somewhere we need to say that the go-pluginserver must be compiled.  maybe something like "the `go-pluginserver` executable, built with the same environment as the plugins.  Kong will expect...."